sled-hardware/src/illumos/sysconf.rs

```diff
@@ -25,14 +25,21 @@ pub fn online_processor_count() -> Result<u32, Error> {
     Ok(u32::try_from(res)?)
 }
 
-/// Returns the amount of RAM on this sled, in bytes.
-pub fn usable_physical_ram_bytes() -> Result<u64, Error> {
-    let phys_pages: u64 = illumos_utils::libc::sysconf(libc::_SC_PHYS_PAGES)
+/// Returns the number of physical RAM pages on this sled.
+pub fn usable_physical_pages() -> Result<u64, Error> {
+    let pages = illumos_utils::libc::sysconf(libc::_SC_PHYS_PAGES)
         .map_err(|e| Error::Sysconf { arg: "physical pages", e })?
         .try_into()?;
+    Ok(pages)
+}
+
+/// Returns the amount of RAM on this sled, in bytes.
+pub fn usable_physical_ram_bytes() -> Result<u64, Error> {
     let page_size: u64 = illumos_utils::libc::sysconf(libc::_SC_PAGESIZE)
         .map_err(|e| Error::Sysconf { arg: "physical page size", e })?
         .try_into()?;
 
-    Ok(phys_pages * page_size)
+    // XXX: if we eventually have pages with mixed sizes, this may be wrong!
+    // I'm not even sure how we'd calculate this in such a world!
+    Ok(usable_physical_pages()? * page_size)
```

Comment:
FWIW the man page for `sysconf(3C)` more or less says in its Notes section that you *ought to* be able to multiply `_SC_PHYS_PAGES` by `_SC_PAGESIZE` to get the total system memory size. (It just says it parenthetically as part of a warning about 32-bit overflow.)

Reply:
this got me to look a bit further into illumos on this, and yeah, that's right. it's a bit funny: installed memory is measured in bytes in `installed_top_size_ex()`, then divided by PAGESIZE (via right shift), and here we'd be collecting the quotient and denominator to get bytes back.

i'll switch to the appropriate `sysconf()` math, thanks.

Reply:
that matches with my understanding, yeah. i've left some notes here in https://github.com/oxidecomputer/omicron/pull/7837/commits/f474252eda8016171b5ceb912522b3e08a01adc7 in case future readers have the same wondering.